[Cocoa, MSE] MSE video doesn't render in WK1 WebView

eric-carlson opened the pull request:
#### ba5d6a18a598b8c8d8e328174609384590c0a3b2
<pre>
[Cocoa, MSE] MSE video doesn&apos;t render in WK1 WebView
<a href="https://bugs.webkit.org/show_bug.cgi?id=250105">https://bugs.webkit.org/show_bug.cgi?id=250105</a>
rdar://103696863

Reviewed by Wenson Hsieh.

252183@main (rdar://96256430) updated the `MediaPlayerPrivateMediaSourceAVFObjC` logic for
rendering with an AVSampleBufferDisplayLayer or with a decompression session. The logic
includes a check of the `mediaSourceInlinePaintingEnabled` runtime flag, which is off by
default in WebKit 1, so it inadvertently disabled video in legacy WebViews.

* LayoutTests/media/media-source/content/test-red-3s-480x360.png: Added.
* LayoutTests/media/media-source/media-source-video-renders-expected.html: Added.
* LayoutTests/media/media-source/media-source-video-renders.html: Added.
* LayoutTests/media/utilities.js:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml: Make the MediaSourceInlinePaintingEnabled
WebKitLegacy default the same as WebKit.

Canonical link: <a href="https://commits.webkit.org/258499@main">https://commits.webkit.org/258499@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6093d7fd06ed95277b509a0cb6b8485489c85a15

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102133 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11278 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35205 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111458 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/171632 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12257 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2191 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94512 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109195 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107912 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9382 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/92660 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/37196 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91251 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24141 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/78922 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [❌ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/92486 "Found 1 new JSC stress test failure: wasm.yaml/wasm/lowExecutableMemory/exports-oom.js.default-wasm (failure)") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4830 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25565 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/88696 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/2464 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4945 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/2003 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/29483 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10999 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45061 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/91616 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5829 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6690 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/20477 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

whsieh reviewed (approved):
r=mews after rebasing against trunk.

eric-carlson committed:
[Cocoa, MSE] MSE video doesn't render in WK1 WebView
https://bugs.webkit.org/show_bug.cgi?id=250105
rdar://103696863

Reviewed by Wenson Hsieh.

252183@main (rdar://96256430) updated the `MediaPlayerPrivateMediaSourceAVFObjC` logic for
rendering with an AVSampleBufferDisplayLayer or with a decompression session. The logic
includes a check of the `mediaSourceInlinePaintingEnabled` runtime flag, which is off by
default in WebKit 1, so it inadvertently disabled video in legacy WebViews.

* LayoutTests/media/media-source/content/test-red-3s-480x360.png: Added.
* LayoutTests/media/media-source/media-source-video-renders-expected.html: Added.
* LayoutTests/media/media-source/media-source-video-renders.html: Added.
* LayoutTests/media/utilities.js:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml: Make the MediaSourceInlinePaintingEnabled
WebKitLegacy default the same as WebKit.

Canonical link: https://commits.webkit.org/258499@main
